File: Swift/relative-ranks.swift
class Solution {
    func findRelativeRanks(_ nums: [Int]) -> [String] {
        let sortedNums = nums.sorted().reversed()
        var ranks = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        ranks = nums.count > 3 ? ranks + (4..<nums.count+1).map{ return String($0) } : ranks
        let dict = Dictionary(uniqueKeysWithValues: zip(sortedNums, ranks))
        return nums.map{ return dict[$0]! }
    }
}

File: Python/relative-ranks.py
class Solution:
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        sortedNums = sorted(nums)[::-1]
        ranks = ['Gold Medal', 'Silver Medal', 'Bronze Medal']
        ranks = ranks + list(map(str, range(4, len(nums) + 1))) if len(nums) > 3 else ranks
        return list(map(dict(zip(sortedNums, ranks)).get, nums))
